power_grid_model_c/power_grid_model/include/power_grid_model/common/exception.hpp

```diff
@@ -166,6 +166,11 @@ class DuplicativelyRegulatedObject : public PowerGridError {
     }
 };
 
+class NotImplementedError : public PowerGridError {
```

Comment:
this needs to be added to `src/power_grid_model/errors.py` and `src/power_grid_model/core/error_handling.py`